File: TastyAppBackend/src/main/java/com/kondziu/projects/TastyAppBackend/dto/CommentDto.java
package com.kondziu.projects.TastyAppBackend.dto;

import lombok.*;

import java.util.Date;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommentDto {

    private String comment;
    private Date date;
    //username is optional
    private Long userId;
    private String username;
}

File: TastyApp/src/app/models/commentDto.ts
export class commentDto{

    constructor(comment:string, userId: number = 0, date: Date = new Date(), username?: string){
        this.comment = comment;
        this.date = date;
        this. userId = userId;
        this.username = username;
    }

    public comment?: string;
    public  date: Date | undefined;
    //username is optional
    public  userId: number;
    public  username: string | undefined;
}